fingerprint_report_templates/fingerprint_report/templates/macros.html

```diff
@@ -58,4 +58,129 @@
     {% else %}
         {{ render_error("How did you get here? did you forget to use 'render_fetch_results' macro?") }}
     {% endif %}
+{%- endmacro %}
+
+{############################################}
+{# fingerpritinging macros #}
+{############################################}
+
+{# select all the graphs is the default endpoint #}
+{% macro select_graphs() -%}
+    {% set select_query =
+    """select distinct ?graph
+    {graph ?graph {?s ?p []}}""" %}
+    {% set content, error = from_endpoint(conf.default_endpoint).with_query(select_query).fetch_tabular() %}
+    {# caller_ is a workaround necessary to handle two levels of callers #}
+    {% set caller_ = caller %}
+    {# Fingerpring the gefault graph before going to named graphs #}
+    {{ caller_(None) }}
+    {% call render_fetch_results(content, error) %}
+        {% if not content.empty %}
+            {% for row in content.itertuples() %}
+                {{ caller_(row.graph) }}
+            {% endfor %}
+        {% endif %}
+    {% endcall %}
+{%- endmacro %}
+
+{# select all the classes is the given graph #}
+{% macro select_classes_in_graph(graph=None) -%}
```

Comment:
`select_distinct_class_types` (I'd not include `graph` in the name as it isn't a required parameter, so calling 
```python 
select_distinct_class_types_in_graph()
``` 
without any parameters will seem like an error